services/jenkins-pipeline-monitor/handler.py

```diff
@@ -0,0 +1,134 @@
+import os
+import boto3
+import json
+import logging
+import secret_manager
+
+from jenkinsapi.jenkins import Jenkins
+
+logging.getLogger().setLevel(logging.INFO)
+logging.getLogger('boto3').setLevel(logging.CRITICAL)
+logging.getLogger('botocore').setLevel(logging.CRITICAL)
+
+
+def get_jenkins_obj(secret):
+    """
+    This method returns an object of Jenkins instantiated using username, password
+    """
+    jenkins_url, jenkins_username, jenkins_password = os.environ["JENKINS_URL"], secret["jenkins_username"], secret["jenkins_password"]
+    return Jenkins(jenkins_url, username=jenkins_username, password=jenkins_password)
+
+
+def get_secret():
+    """
+    This method is to get secret value from Secrets Manager
+    """
+    secret = json.loads(secret_manager.get_secret())
+    return secret
+
+
+def get_pipeline_job(jenkinsObj):
+    job = jenkinsObj["restricted-mxnet-cd/mxnet-cd-release-job"]
+    return job
+
+
+def get_latest_build_number(job):
+    return job.get_last_build().get_number()
+
+
+def get_build_from_build_number(job, build_number):
+    return job.get_build(build_number)
+
+
+def get_build_timestamp(build):
+    return build.get_timestamp()
+
+
+def get_build_date(timestamp):
+    return timestamp.date()
+
+
+def is_latest_day_build(current_build, latest_build):
+    current_build_timestamp = get_build_timestamp(current_build)
+    latest_build_timestamp = get_build_timestamp(latest_build)
+    # if 2 builds are within 24 hours and on the same day
```

Comment:
why two conditions? isn't "on the same day" enough?

Reply:
previously i was thinking timezone difference might cause few issues for eg

5pm PT would be next day in UTC
and hence the day would be different. In that case we would have to check for 24hours.

Right now, I have put 11:59pm UTC which ensures same day check is enough. I'll remove the 24hour diff check.

Reply:
No, never make timing assumptions. Please restore the check.

Reply:
Yeah let’s keep last 24 hours check instead of same day check. That should work irrespective of the time zone.

Reply:
So we are May 9 11:59pm UTC [lambda is triggered]

Latest build as of May 9 11:59pm UTC is _Build 1088_ [May 7 9:54:39pm]

http://jenkins.mxnet-ci.amazon-ml.com/job/restricted-mxnet-cd/job/mxnet-cd-release-job/1088/

#### If we relax the same day check & only keep the 24hour check.
So all builds after May 6 9:55:39 pm will be accepted.
1082 to 1088


Here 1082 isn't supposed to be selected as it was part of the previous day trigger
_Build 1082_ May 6, 2020, 9:57:08 PM [which is not supposed to be picked]
http://jenkins.mxnet-ci.amazon-ml.com/job/restricted-mxnet-cd/job/mxnet-cd-release-job/1082/

#### Hence we need both checks

Reply:
As per offline discussion, checking for last 6 hours would suffice.

Reply:
Addressed.
https://github.com/apache/incubator-mxnet-ci/pull/25/commits/2da87ac0424dd68b799cd24ded81e27ebc3a0969

Reply:
I really don't get your guys here. This results in the report swallowing 2/3 of a day as well as additional runs, leaving issues unnoticed. 

To me this change (and the motivation) doesn't make sense.

Reply:
To clarify the motivation, here we are concerned with monitoring CD PyPI packages distribution, which specifically refers to two pipelines (mxnet_lib/static and pypi_packages) that run daily in a fixed time interval one after the other. The lambda function searches for these two pipelines by using the author and time filters.

Reply:
Well, but once the timing changes, this script will be broken. So why set an arbitrary timeframe like 8h instead of simply having a full day and generating the report every 24h? The timeframe and the schedule have to be identical to not produce a gap.

Reply:
Reverting back the 24hr requirement instead of arbitrary 8hr one.
Refer : https://github.com/apache/incubator-mxnet-ci/pull/26
Commit : https://github.com/apache/incubator-mxnet-ci/pull/26/commits/4bbe05c6d7da33c66569ff388f4d5265806687f2